raiden/transfer/mediated_transfer/initiator.py

```diff
@@ -560,7 +560,7 @@ def handle_send_secret_reveal_light(
 
     initiator_state.revealsecret = revealsecret
     initiator_state.received_secret_request = True
-    receiver_light_client_address = state_change.receiver if channel_state.both_participants_are_light_clients else None
+    receiver_light_client_address = state_change.receiver
```

Comment:
This cannot be always the transfer.initiator because on payments between LC and FN the initiator can be a not registered client, therefore the client FK will fail